src/source/Ice/IceAgent.c

```diff
@@ -2151,9 +2190,18 @@ STATUS incomingDataHandler(UINT64 customData, PSocketConnection pSocketConnectio
     // for stun packets, first 8 bytes are 4 byte type and length, then 4 byte magic byte
     if ((bufferLen < 8 || !IS_STUN_PACKET(pBuffer)) && pIceAgent->iceAgentCallbacks.inboundPacketFn != NULL) {
         // release lock early
+        retStatus = findIceCandidatePairWithLocalSocketConnectionAndRemoteAddr(pIceAgent, pSocketConnection, pSrc, TRUE, &pIceCandidatePair);
         MUTEX_UNLOCK(pIceAgent->lock);
         locked = FALSE;
         pIceAgent->iceAgentCallbacks.inboundPacketFn(pIceAgent->iceAgentCallbacks.customData, pBuffer, bufferLen);
+        if (pIceCandidatePair != NULL) {
+            if (pIceCandidatePair == pIceAgent->pDataSendingIceCandidatePair) {
```

Comment:
why not just compare pDataSendingIceCandidatePair with pSocketConnection and remote address?

Reply:
Agreed. Fixed.